services/local-notifications/core.go

```diff
@@ -0,0 +1,322 @@
+	Value       *hexutil.Big      `json:"value"`
+	ERC20       bool              `json:"erc20"`
+	Contract    common.Address    `json:"contract"`
+	Network     uint64            `json:"network"`
+}
+
+type Notification struct {
+	ID            common.Hash      `json:"id"`
+	Platform      float32          `json:"platform,omitempty"`
+	Body          notificationBody `json:"body"`
+	Category      PushCategory     `json:"category,omitempty"`
+	Image         string           `json:"imageUrl,omitempty"`
+	IsScheduled   bool             `json:"isScheduled,omitempty"`
+	ScheduledTime string           `json:"scheduleTime,omitempty"`
+}
+
+// TransactionEvent - structure used to pass messages from wallet to bus
+type TransactionEvent struct {
+	Type                      string                 `json:"type"`
+	BlockNumber               *big.Int               `json:"block-number"`
+	Accounts                  []common.Address       `json:"accounts"`
+	NewTransactionsPerAccount map[common.Address]int `json:"new-transactions"`
+	ERC20                     bool                   `json:"erc20"`
+}
+
+// MessageEvent - structure used to pass messages from chat to bus
+type MessageEvent struct{}
+
+// CustomEvent - structure used to pass custom user set messages to bus
+type CustomEvent struct{}
+
+const topic = "local-notifications"
+
+type transmitter struct {
+	wg        sync.WaitGroup
+	publisher *event.Feed
+	quit      chan struct{}
+}
+
+// Service keeps the state of message bus
+type Service struct {
+	started           bool
+	transmitter       *transmitter
+	walletTransmitter *transmitter
+	db                *Database
+	walletDB          *wallet.Database
+	accountsDB        *accounts.Database
+}
+
+func NewService(appDB *sql.DB, network uint64) *Service {
+	db := NewDB(appDB, network)
+	walletDB := wallet.NewDB(appDB, network)
+	accountsDB := accounts.NewDB(appDB)
+	trans := &transmitter{}
+	walletTrans := &transmitter{}
+
+	return &Service{
+		db:                db,
+		walletDB:          walletDB,
+		accountsDB:        accountsDB,
+		transmitter:       trans,
+		walletTransmitter: walletTrans,
+	}
+}
+
+func pushMessage(notification *Notification) {
+	log.Info("Pushing a new push notification", "info", notification)
+	signal.SendLocalNotifications(notification)
+}
+
+func (s *Service) buildTransactionNotification(rawTransfer wallet.Transfer) *Notification {
+	log.Info("Handled a new transfer in buildTransactionNotification", "info", rawTransfer)
+
+	var state = undetermined
+	transfer := wallet.CastToTransferView(rawTransfer)
+
+	switch {
+	case transfer.TxStatus == hexutil.Uint64(0):
+		state = failed
+	case transfer.Address == transfer.To:
+		state = inbound
+	default:
+		state = outbound
+	}
+
+	from, err := s.accountsDB.GetAccountByAddress(types.Address(transfer.From))
+
+	if err != nil {
+		log.Debug("Could not select From account by address", "error", err)
+	}
+
+	to, err := s.accountsDB.GetAccountByAddress(types.Address(transfer.To))
```

Comment:
We use different types for the address in the Multiaccount module and the Wallet, implementation is the same as seems like the one used in `"github.com/status-im/status-go/eth-node/types"` is a fork from `"github.com/ethereum/go-ethereum/common"`. IMO we should use everywhere the `go-ethereum` types